### ISSUE TITLE
[Panel] Add button still visible if user doesn't have `create` permission.

### ISSUE THREAD

**Issue by texnixe** [getkirby/kirby#1855]
**Describe the bug**
When setting `create: false` in the page blueprint options, the "Add" button is still visible and the user has to fill in the create form before getting the message that they are not allowed to do so.

```

options:
  create:
    *: false
    admin: true
```

**To Reproduce**
1. Create a page blueprint with the above setting (or just `create: false`) and log in as non-admin (or any) user
2. Visit a page with the modified blueprint. The "Add" button is still there and can be clicked.
3. Only after user fills in form, a "no no" message appears

**Expected behavior**
The "Add" button should be removed/disabled, so that the user doesn't have to fill in the create form just to learn that they are not allowed to do so.

**Kirby Version**
3.2rc2

**Comment by afbora** [comment]
Actually, I think it's gonna be a little difficult. 
Because you know that when we click the add button, we first select the page template. 
So we need to check all the allowed templates for the add button and the templates that can be selected in the create dialog.

**Comment by distantnative** [comment]
I think the confusion might be about to what page the option applies:
When `notes.yml` features the option as mentioned by @texnixe, I assume it means that only admins are allowed to create new `notes.yml`-based pages. Not that only admins can add subpages to `notes.yml`.

As @afbora mentioned, we would have on eahc page load to check all the allowed templates to see whether none of them can be created to only then hide the add button. Difficult.


`create` is not documented btw: https://getkirby.com/docs/reference/panel/blueprints/page#options

**Comment by bastianallgeier** [comment]
This is indeed difficult without adding a serious performance penalty to the pages section. I think what we could do is to either lazy-load the add option in a second request from the API once the pages section has been loaded, but this means that the add button will flickr and appear a bit later then the rest. Another option would be that we cache the permissions and all the available blueprints so this is only an issue on the first request. I'm afraid we will need to move this to the next milestone again :(

**Comment by francoismajor** [comment]
I would also expect if some templates are allowed and others not, that the ones not allowed don't show up as options to create, or at worst are disabled. (from the dropdown list). Right now they are all visible and only when trying to save the dialog do you get an error.

Another problem:
This bug is true in the reverse situation.

If I set permissions to create pages to false for a user role
```
[editor]
  pages:
    create: false
```

and then override them on a per page basis

```
[myPage]
options:
  create:
    editor: true
```

the Add button does not show up, even though I'm allowed to create such a page.

I understand the complexity and performance issues, but this does limit quite a bit what you can do with the permissions and also leads to some unintuitive interface problems.

Brainstorming here. How about if we move the grunt work off to the dev? rather than trying to calculate it dynamically. We could move/add the 'permissions' definition up one level.

So something like this
```
myPage
sections:
      subPage:
        type: pages
        status: all
        templates:
          - subPage
               options:
                 create:
                   editor: true
```
You could still define them as before in users and in pages, but adding this option would allow to easily show only the allowed templates/Add button, or none at all if set to false, without performance hits. 
Downside is more work for the dev (meh) and maybe having to write permissions in multiple places.

**Comment by distantnative** [comment]
@bastianallgeier I think we might have a stab at this if we go the Inertia path and send the section data with the first request. Cause then I think even the heavy lifting of going through all blueprints might be tolerable... well, maybe. 

Otherwise, this is really a design issue and it would better to define on a parent who is allowed to create subpages. Instead on the page-level itself who can create such a page.

**Comment by jonasfeige** [comment]
I just ran into this issue as well. However, it does not seem to be only about the Add button being visible or not, but the `create` option not having any effect at all when set anywhere but a within a user blueprint.

I just tried the following with a fresh Starterkit (3.5.3):

site.yml
```
options:
  create:
    *: false
    admin: true
```

While logged in as an `editor` (with no specific permissions set in the user blueprint) I am still able to create pages.

**Comment by Warchamp7** [comment]
Ran into this issue today. Trying to create a role that is only able to manage pages in a subsection of our website. Is the creation of pages still limited to a global on/off when it comes to blueprints?

I basically want a role that can't create pages site-wide, but then within a specific page blueprint override it to allow it.

The above noted performance issue sounds like it's from the context of `create: true` being specified in a blueprint, of allowing that particular template to be created.

Rather, I propose `create` to control whether subpages in that template can be made.

The `templates` field itself can then control what templates are available. Either globally (As it exists now) or on a per role basis

Ex. in a `blog` blueprint I currently have
```
posts:
  headline: Blog Posts
  type: pages
  status: all
  templates: blogpost
```

This could be extended to support
```
posts:
  headline: Blog Posts
  type: pages
  status: all
    templates:
      manager:
        blogpost
        otherpost
      blogwriter: blogpost
```

and `create: true` could be set as an option on this `blog` blueprint

**Comment by lukasbestle** [comment]
@Warchamp7 This is a bit of a different issue. Here it's about the add button in particular, meaning the permissions are already handled, but the add button is still displayed and clicking on it will directly result in an error message. Your use-case is more about new options to actually limit permissions per role, not just to disable the add button.

Limiting templates per role from the blueprint is currently not supported, but I think it's an interesting feature idea. Please add it to our feedback platform (https://feedback.getkirby.com) to avoid that the idea gets lost in this issue.

Until such a feature is available, you can use a [`page.create:before` hook](https://getkirby.com/docs/reference/plugins/hooks/page-create-before) to dynamically prevent the creation of pages for specific roles. It's not as clean, but it could be a workaround for your use-case until an easier solution is available.

**Comment by Warchamp7** [comment]
> @Warchamp7 This is a bit of a different issue. Here it's about the add button in particular, meaning the permissions are already handled, but the add button is still displayed and clicking on it will directly result in an error message. Your use-case is more about new options to actually limit permissions per role, not just to disable the add button.
> 
> Limiting templates per role from the blueprint is currently not supported, but I think it's an interesting feature idea. Please add it to our feedback platform (https://feedback.getkirby.com) to avoid that the idea gets lost in this issue.
> 
> Until such a feature is available, you can use a [`page.create:before` hook](https://getkirby.com/docs/reference/plugins/hooks/page-create-before) to dynamically prevent the creation of pages for specific roles. It's not as clean, but it could be a workaround for your use-case until an easier solution is available.

Yeah, I'm using a hook right now to enforce the restrictions I want, but it doesn't seem like I can control the Add button being available to a role on a per blueprint basis.

Perhaps I'm doing something incorrect, but if I set `create: false` in a role blueprint, is there no way to override that in a page blueprint for that role?

**Comment by lukasbestle** [comment]
> but it doesn't seem like I can control the Add button being available to a role on a per blueprint basis.

That's correct, the hook is only run once the user actually tries to create a page, not in advance.

> Perhaps I'm doing something incorrect, but if I set `create: false` in a role blueprint, is there no way to override that in a page blueprint for that role?

I don't think so. @distantnative?

**Comment by Warchamp7** [comment]
> > but it doesn't seem like I can control the Add button being available to a role on a per blueprint basis.
> 
> That's correct, the hook is only run once the user actually tries to create a page, not in advance.
> 
> > Perhaps I'm doing something incorrect, but if I set `create: false` in a role blueprint, is there no way to override that in a page blueprint for that role?
> 
> I don't think so. @distantnative?

Yeah, I understand the hook system. It's how I had this sort of restriction in place on Kirby2

For now, almost all places I want to restrict page creation by role is subpages of a "container" page, so I've just locked out read access to the container pages, have the container child template restricted to only the allowed templates, and have a hook in place as a failsafe on top of it all.

I think my above proposal would be a low impact way to solve this scenario

**Comment by distantnative** [comment]
Let's try to split these discussion again:

1. Visibility of "Add" button: section would need to parse all allowed templates to see if the current user is allowed to create any/none of them and show/hide the button accordingly. @bastianallgeier maybe something we can look into with Fiber sections?

2. Allow global role permissions to be overwritten by blueprint-specific ones: tbh I am not sure what the current state is there, but @Warchamp7 it would be great if you could open a separate issue for that. Thanks!
